app/controllers/secondary_content_links_controller.rb

```diff
@@ -33,21 +33,43 @@ def destroy
 
       redirect_to step_by_step_page_secondary_content_links_path(step_by_step_page.id), notice: 'Secondary content link was successfully deleted.'
     else
-      redirect_to step_by_step_page_secondary_content_links_path
+      redirect_to step_by_step_page_secondary_content_links_path(step_by_step_page.id)
     end
   end
 
+  def index
+    step_by_step_page
+    secondary_content_links
+    @new_secondary_content_link = new_secondary_content_link
+  end
+
 private
 
   def step_by_step_page
     @step_by_step_page = StepByStepPage.find(params[:step_by_step_page_id])
   end
 
+  def secondary_content_links
+    @secondary_content_links = StepByStepPage.find(params[:step_by_step_page_id]).secondary_content_links
+  end
+
+  def new_secondary_content_link
+    step_by_step_page.secondary_content_links.new
+  end
+
   def secondary_content_link
     @secondary_content_link ||= step_by_step_page.secondary_content_links.find(params[:id])
   end
 
   def update_downstream
     StepByStepDraftUpdateWorker.perform_async(step_by_step_page.id, current_user.name)
   end
+
+  def step_params
```

Comment:
@DilwoarH Where is this used?

Reply:
Once we have the `base_path` we can get the `title` and `content_id` from publishing-api, so we might want to move all of this to the model.

Reply:
Yeah - this was just done as a placeholder, your function expected these params to be passed in, so we had to pass it in to make everything work.

Reply:
@leenagupte - this is used here https://github.com/alphagov/collections-publisher/pull/618/files#diff-49365286d564b9a2a3a2a422330233deR9